tests/components/onkyo/test_config_flow.py

```diff
@@ -20,12 +20,60 @@
 
 from . import (
     create_config_entry_from_info,
+    create_connection,
     create_empty_config_entry,
     create_receiver_info,
     setup_integration,
 )
 
-from tests.common import Mock, MockConfigEntry
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(autouse=True)
+def patch_timeouts():
+    """Patch timeouts to avoid tests waiting."""
+    with (
+        patch(
+            "homeassistant.components.onkyo.receiver.DEVICE_INTERVIEW_TIMEOUT", new=0
+        ),
+        patch(
+            "homeassistant.components.onkyo.receiver.DEVICE_DISCOVERY_TIMEOUT", new=0
+        ),
+    ):
+        yield
+
+
+@pytest.fixture
+async def default_mock_discovery():
+    """Mock discovery with a single device."""
+
+    async def mock_discover(host=None, discovery_callback=None, timeout=0):
+        await discovery_callback(create_connection(1))
+
+    with patch("pyeiscp.Connection.discover", new=mock_discover):
```

Comment:
Ideally we patch the library where we use it, so `homeassistant.components.onkyo.*`